hal/src/core/pinmap_impl.h

```diff
@@ -36,6 +36,7 @@ extern "C" {
 typedef struct STM32_Pin_Info {
   GPIO_TypeDef* gpio_peripheral;
   pin_t gpio_pin;
+  uint8_t gpio_pin_source;
```

Comment:
This must come at the end of the pinmap or older applications using the old version of pinmap will crash if newer system firmware is updated.